graphix/simulator.py

```diff
@@ -6,29 +6,77 @@
 
 from __future__ import annotations
 
+import abc
 import warnings
 
 import numpy as np
 
-from graphix.command import CommandKind
+import graphix.clifford
+from graphix.command import BaseM, CommandKind, M
+from graphix.pauli import MeasureUpdate
+from graphix.sim.base_backend import Backend, MeasurementDescription
 from graphix.sim.density_matrix import DensityMatrixBackend
 from graphix.sim.statevec import StatevectorBackend
 from graphix.sim.tensornet import TensorNetworkBackend
 
 
+class MeasureMethod(abc.ABC):
+    def measure(self, backend: Backend, cmd, noise_model=None) -> None:
+        description = self.get_measurement_description(cmd)
+        result = backend.measure(cmd.node, description)
+        if noise_model is not None:
+            result = noise_model.confuse_result(result)
+        self.set_measure_result(cmd.node, result)
+
+    @abc.abstractmethod
+    def get_measurement_description(self, cmd: BaseM) -> MeasurementDescription: ...
+
+    @abc.abstractmethod
+    def get_measure_result(self, node: int) -> bool: ...
+
+    @abc.abstractmethod
+    def set_measure_result(self, node: int, result: bool) -> None: ...
+
+
+class DefaultMeasureMethod(MeasureMethod):
```

Comment:
Could you add description of this class, esp. why we need to have this in the first place? I believe this is for generalization (e.g. client-server setup), but it is hard to understand why we have this without such context, as this looks (at first sight) a rather unnecessary wrapping of measurement command.

Reply:
It is exactly for that reason (see 4th point in the description of the PR): this class is to be overwritten by a custom measurement method in the case of a client-server setup. You are right we should add a comment about it in the code.

Reply:
thanks! this kind of targeted API change would require a simple but complete example demonstrating such use and/or a dedicated tutorial page. could you write up an example?

Reply:
Thanks! tbh, I feel that delegating the explanation and justification of implementation decisions to the package that depends on graphix is a little strange. it is like rewriting some part of numpy for the sake of graphix and justifying by linking graphix example. I would like this repo to be standalone to be maintainable, and I would appreciate it if you could add short example in graphix package without relying on verphix if at all possible.

Reply:
@jemappellesami @thierry-martinez I guess I'll write one up myself..! let us merge this now and we can get on with other PRs!